src/recoilState/appState.ts

```diff
@@ -500,3 +505,70 @@ export const rUserGifts = selectorFamily<
 });
 export const useUserGifts = (userId: number) =>
   useRecoilValue(rUserGifts(userId));
+
+export type IReceivedGiftsByUser = { [account: string]: number | string };
+
+// Todo: Fetch this data from API when available
+export const rUnclaimedTokensByUser = atom<Record<string, number>>({
```

Comment:
the data we need to add to the current merkle root is not just the amount of unclaimed tokens -- it's actually the entire merkle root for the previous epoch. so the code would be something like this:

```js
const previousRoot = fetchPreviousEpochRoot(); // TODO from API or subgraph
const newRoot = parseBalanceMap(receivedGiftsByUser); // only the ones from the current epoch

// this is what you need to implement, and what gets sent to the contract
const combinedRoot = combineRoots(previousRoot, newRoot); 
```